src/ServiceJobList.c

```diff
@@ -0,0 +1,143 @@
+    g_messages_ListJobs.response.names.size = 0;
+
+    MOTOROS2_MEM_TRACE_REPORT(svc_list_jobs_init);
+}
+
+void Ros_ServiceListJobs_Cleanup()
+{
+    rcl_ret_t ret;
+    MOTOROS2_MEM_TRACE_START(svc_list_jobs_fini);
+
+    Ros_Debug_BroadcastMsg("Cleanup service " SERVICE_NAME_LIST_JOBS);
+    ret = rcl_service_fini(&g_serviceListJobs, &g_microRosNodeInfo.node);
+    if (ret != RCL_RET_OK)
+        Ros_Debug_BroadcastMsg("Failed cleaning up " SERVICE_NAME_LIST_JOBS " service: %d", ret);
+    rosidl_runtime_c__String__fini(&g_messages_ListJobs.response.message);
+    rosidl_runtime_c__String__Sequence__fini(&g_messages_ListJobs.response.names);
+
+    MOTOROS2_MEM_TRACE_REPORT(svc_list_jobs_fini);
+}
+
+void Ros_ServiceListJobs_Trigger(const void* request_msg, void* response_msg)
+{
+    RCL_UNUSED(request_msg);
+    ListJobsResponse* response = (ListJobsResponse*)response_msg;
+
+    Ros_Debug_BroadcastMsg("%s: enter", __func__);
+
+    //make sure we don't accidentally return old results (in case of errors fi)
+    rosidl_runtime_c__String__assign(&response->message, "");
+    response->result_code = 0;
+    response->names.size = 0;
+
+    //retrieve list of jobs (".JBI")
+    //TODO(gavanderhoorn): should this also look for other files (JBR)?
+    Ros_Debug_BroadcastMsg("%s: updating job file list", __func__);
+    LONG status = mpRefreshFileList(MP_EXT_ID_JBI);
+    if (status != OK)
+    {
+        Ros_Debug_BroadcastMsg("%s: error refreshing job list", __func__);
+        rosidl_runtime_c__String__assign(&response->message, "error refreshing job list");
+        //TODO(gavanderhoorn): define status/error codes
+        response->result_code = 10;
+        goto DONE;
+    }
+
+    //see how many jobs there are, and if necessary, resize list of names in service response
+    size_t job_count = mpGetFileCount();
+    if (job_count < 0)
+    {
+        Ros_Debug_BroadcastMsg("%s: error retrieving file count: %d", __func__, job_count);
+        rosidl_runtime_c__String__assign(&response->message, "error retrieving file count");
+        //TODO(gavanderhoorn): define status/error codes
+        response->result_code = 11;
+        goto DONE;
+    }
+    Ros_Debug_BroadcastMsg("%s: found %d jobs", __func__, job_count);
+    if (job_count > response->names.capacity)
+    {
+        Ros_Debug_BroadcastMsg("%s: resize needed: current: %d; need: %d",
+            __func__, job_count, response->names.capacity);
+        rosidl_runtime_c__String__Sequence__fini(&response->names);
+        //TODO(gavanderhoorn): handle errors
+        rosidl_runtime_c__String__Sequence__init(&response->names, job_count);
+    }
+
+    //copy all jobs on the list to the response
+    //32 characters for the name, plus 4 for extension, plus terminating null
+    //TODO(gavanderhoorn): is this really necessary? Copied from inform generator
+    //TODO(gavanderhoorn): confirm whether extension is included or not
```

Comment:
`mpGetFileName(..)` does appear to include the job extension.

@ted-miller: if we're only going to support `JBI` management, could we perhaps remove the extension?

Reply:
Sure